src/components/IstioWizards/IstioWizardActions.ts

```diff
@@ -389,63 +392,101 @@ export const buildIstioConfig = (
-        wizardDR.spec.trafficPolicy.tls.clientCertificate = wState.trafficPolicy.clientCertificate;
-        wizardDR.spec.trafficPolicy.tls.privateKey = wState.trafficPolicy.privateKey;
-        wizardDR.spec.trafficPolicy.tls.caCertificates = wState.trafficPolicy.caCertificates;
-      }
+    if (wState.trafficPolicy.mtlsMode === MUTUAL) {
+      wizardDR.spec.trafficPolicy.tls.clientCertificate = wState.trafficPolicy.clientCertificate;
+      wizardDR.spec.trafficPolicy.tls.privateKey = wState.trafficPolicy.privateKey;
+      wizardDR.spec.trafficPolicy.tls.caCertificates = wState.trafficPolicy.caCertificates;
     }
-    if (wState.trafficPolicy.addLoadBalancer) {
-      if (wState.trafficPolicy.simpleLB) {
-        // Remember to put a null fields that need to be deleted on a JSON merge patch
-        wizardDR.spec.trafficPolicy.loadBalancer = {
-          simple: wState.trafficPolicy.loadBalancer.simple,
-          consistentHash: null
-        };
-      } else {
-        wizardDR.spec.trafficPolicy.loadBalancer = {
-          simple: null,
-          consistentHash: {}
-        };
-        wizardDR.spec.trafficPolicy.loadBalancer.consistentHash = {
-          httpHeaderName: null,
-          httpCookie: null,
-          useSourceIp: null
-        };
-        if (wState.trafficPolicy.loadBalancer.consistentHash) {
-          const consistentHash = wState.trafficPolicy.loadBalancer.consistentHash;
-          switch (wState.trafficPolicy.consistentHashType) {
-            case ConsistentHashType.HTTP_HEADER_NAME:
-              wizardDR.spec.trafficPolicy.loadBalancer.consistentHash.httpHeaderName = consistentHash.httpHeaderName;
-              break;
-            case ConsistentHashType.HTTP_COOKIE:
-              wizardDR.spec.trafficPolicy.loadBalancer.consistentHash.httpCookie = consistentHash.httpCookie;
-              break;
-            case ConsistentHashType.USE_SOURCE_IP:
-              wizardDR.spec.trafficPolicy.loadBalancer.consistentHash.useSourceIp = true;
-              break;
-            default:
-            /// No default action
+
+    if (wState.trafficPolicy.peerAuthnSelector.addPeerAuthentication) {
+      const peerAuthnLabels: { [key: string]: string } = {};
+      peerAuthnLabels[serverConfig.istioLabels.appLabelName] = wProps.workloads[0].labels![
+        serverConfig.istioLabels.appLabelName
+      ];
+
+      wizardPA = {
+        metadata: {
+          namespace: wProps.namespace,
+          name: wProps.serviceName,
+          labels: {
+            [KIALI_WIZARD_LABEL]: wProps.type
+          }
+        },
+        spec: {
+          selector: {
+            matchLabels: peerAuthnLabels
+          },
+          mtls: {
+            mode: wState.trafficPolicy.peerAuthnSelector.mode
           }
         }
+      };
+
+      wizardDR.metadata.annotations = {};
+      wizardDR.metadata.annotations[KIALI_RELATED_LABEL] =
+        'PeerAuthentication/' + wProps.serviceName + '/' + wizardPA.spec.mtls!.mode;
```

Comment:
I doubt about this one.
I like to link the DR with the PA and the serviceName, that makes sense.
But, if the user modifies the PA the status will fail because is coded there.

I think that PA should be fetched in the ServiceDetails page using the kiali_wizard label and this link with the service, but if I change the status in the Istio Config PA, that should be propagated in the wizard.

It happens today with VS/DR, if you edit them and go back to the service details and open the Wizard, this is updated with this new info.

We can review more in details in you like.